src/Marten/StoreOptions.cs

```diff
@@ -81,61 +87,112 @@ public partial class StoreOptions: IReadOnlyStoreOptions, IMigrationLogger
 
     public StoreOptions()
     {
-        EventGraph = new EventGraph(this);
+        _eventGraph = new EventGraph(this);
         Schema = new MartenRegistry(this);
-        Storage = new StorageFeatures(this);
+        _storage = new StorageFeatures(this);
+
+        _providers = new ProviderGraph(this);
+        _advanced = new AdvancedOptions(this);
+
+        _projections = new ProjectionOptions(this);
+
+        _linq = new LinqParsing(this);
 
-        Providers = new ProviderGraph(this);
-        Advanced = new AdvancedOptions(this);
+        var strategy = new ResiliencePipelineBuilder().AddRetry(new()
+        {
+            ShouldHandle = new PredicateBuilder().Handle<NpgsqlException>().Handle<MartenCommandException>(),
+            MaxRetryAttempts = 3,
+            Delay = TimeSpan.FromMilliseconds(50),
+            BackoffType = DelayBackoffType.Exponential
+        }).Build();
+
+        ResiliencePipeline = strategy;
+    }
 
-        Projections = new ProjectionOptions(this);
+    /// <summary>
+    /// Configure and override the Polly error handling policies for this DocumentStore
+    /// </summary>
+    /// <param name="configure"></param>
+    public void ConfigurePolly(Action<ResiliencePipelineBuilder> configure)
+    {
+        var builder = new ResiliencePipelineBuilder();
+        configure(builder);
 
-        Linq = new LinqParsing(this);
+        ResiliencePipeline = builder.Build();
     }
 
-    internal IList<Type> CompiledQueryTypes { get; } = new List<Type>();
+    internal IList<Type> CompiledQueryTypes => _compiledQueryTypes;
+
+    /// <summary>
+    /// Polly policies for retries within Marten command execution
+    /// </summary>
+    public ResiliencePipeline ResiliencePipeline
```

Comment:
**QUESTION:** Shouldn't we have a single entrypoint for setting up the Resilience Policy? Isn't `ConfigurePolly`  enough?

Reply:
I don't have a strong opinion here. I thought some folks might want to push it in themselves. My guess is that very few people ever take advantage of this feature

Reply:
Changed.